server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStorePinnedTimestampsIT.java

```diff
@@ -197,6 +196,97 @@ public void onFailure(Exception e) {
         remoteStorePinnedTimestampService.rescheduleAsyncUpdatePinnedTimestampTask(TimeValue.timeValueMinutes(3));
     }
 
+    public void testPinExceptionsOlderTimestamp() throws InterruptedException {
+        prepareCluster(1, 1, INDEX_NAME, 0, 2);
+        ensureGreen(INDEX_NAME);
+
+        RemoteStorePinnedTimestampService remoteStorePinnedTimestampService = internalCluster().getInstance(
+            RemoteStorePinnedTimestampService.class,
+            primaryNodeName(INDEX_NAME)
+        );
+
+        CountDownLatch latch = new CountDownLatch(1);
+        remoteStorePinnedTimestampService.pinTimestamp(1234L, "ss1", new LatchedActionListener<>(new ActionListener<>() {
+            @Override
+            public void onResponse(Void unused) {
+                // We expect onFailure to be called
+                fail();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertTrue(e instanceof IllegalArgumentException);
+            }
+        }, latch));
+
+        latch.await();
+    }
+
+    // This test fails as we can't control actual upload of pinned timestamp file. We ideally need a BlobStoreRepository
+    // which can control the speed of upload.
```

Comment:
We can use `REPOSITORIES_SLOWDOWN_SETTING` to actually induce a sleep while pinning the timestamp for segment repo . We can refer to `testPinnedTimestampFailSnapshot` , where we induce a failure through a dynamic repo setting .

Reply:
Made the suggest change. Quite useful, thanks!